### PR TITLE
created accordion component with animation

### PR DESCRIPTION
**Pull request by lukewarmtemp**
#31 

Couldn't figure out initialOpen with the current way I'm doing the animations, but I can revisit it's something important that we really need. Accordion opens both with onClick() and whenever the toggle prop value changes. Note: open/closed state is independent of what the value is set to. Can make an adjustment to open/close corresponding to toggle=true/false but some changes also need to be made in onClick() function to ensure that we don't have a case where the accordion is closed but toggle is set to true and vice versa.

### CODE REVIEW

**Comment by lukewarmtemp** [comment]
Also, I'll get the images of what it looks/prove that it's working sometime tomorrow

File: client/package.json [hunk 1]
```diff
@@ -16,12 +16,18 @@
     "*.{js, jsx, css, scss}": "prettier --write"
   },
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^6.1.1",
```
Comment (jameskokoska) [inline client/package.json:19]: For icons, download the svg you want to use and import it
It reduces package count

File: client/src/components/Accordion/SingleAccordion/SingleAccordion.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { React, useState, useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faAngleDown } from '@fortawesome/free-solid-svg-icons';
+import './SingleAccordion.scss';
+
+const SingleAccordion = ({ header, children, toggle }) => {
```
Comment (jameskokoska) [inline client/src/components/Accordion/SingleAccordion/SingleAccordion.jsx:7]: For initial value, pass in a prop `isInitiallyOpened`
and in the line below do  `const [isActive, setActive] = useState(isInitiallyOpened ? 'active' : 'closed');` for example
This initially opened might be needed for UX to show users that the accordion can be opened (by leaving one opened by default)

**Review by jameskokoska — CHANGES_REQUESTED** [review]
Looks good though!

**Comment by lukewarmtemp** [comment]
Removed the packages for font-awesome and downloaded the SVG for the arrow button. Made it such that the isOpen state and setIsOpen needs to passed from the parent function as props. Makes the code cleaner in the component and serves as a dual purpose for isInitallyOpen because we can pass isOpen = true/false.

**Review by jameskokoska — DISMISSED** [review]
Looks good!

File: client/src/components/Accordion/SingleAccordion/SingleAccordion.jsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    setHeight(isOpen ? `${content.current.scrollHeight}px` : '0px');
+    setRotate(isOpen ? 'accordIcon rotate' : 'accordIcon');
+  }, [isOpen]);
+
+  function toggleAccordion() {
+    setIsOpen(!isOpen);
+  }
+
+  return (
+    <div className="accordian" onClick={toggleAccordion}>
```
Comment (Cali0707) [inline client/src/components/Accordion/SingleAccordion/SingleAccordion.jsx:22]: fix the class name to be spelled properly

File: client/src/components/Accordion/SingleAccordion/SingleAccordion.scss [hunk 4]
```diff
@@ -0,0 +1,31 @@
+@import '../../../scssStyles/mixins';
+
+.accordian {
```
Comment (Cali0707) [inline client/src/components/Accordion/SingleAccordion/SingleAccordion.scss:3]: fix classname to be spelled properly

**Review by Cali0707 — APPROVED** [review]
Looks good, there are two minor spelling mistakes, but I don't think those really matter.

**Pull request merged**